### ISSUE TITLE
ForceCode buggy when creating a new vs code workspace 

### ISSUE THREAD

**Issue by julesx** [daishi4u/ForceCode#25]
i have a script that creates a new vs code workspace for me.  it creates a new directory, checks out a bunch of files, copies in force.json, and creates a vs code workspace file, then opens it.

on that very first vs code launch, it seems like ForceCode is not initialized (no status bar items are present)

i press ctrl-shift-p to invoke command palette
select "open forcecode menu"
i am prompted for my SFDC username (which is already in force.json)
then production or sandbox (which is already in force.json)
then whether or not to automatically deploy on save (which is already in force.json)

then i get following error (image):
![image](https://user-images.githubusercontent.com/3010047/42965415-eb5ed0e4-8b67-11e8-910d-3a6a43474b68.png)

still no status bar items

i then restart vs code and everything works as expected.  upon start i get the prompt asking me if i'd like to login to my org.

update: i am also presented with this error upon vs code restart:
![image](https://user-images.githubusercontent.com/3010047/42966973-c259e3dc-8b6c-11e8-9927-b5f596c91ee1.png)
cant seem to replicate it consistently

**Comment by daishi4u** [comment]
I have fixed the second error and it will be in the next release (The invalid type error...I reported it to myself in an issue I created...) You should send a screenshot of your directory layout. Did you know, you don't need to create the force.json or anything...Simply go into an empty directory and hit Ctrl+Shift+P and then run the ForceCode: ForceCode Menu command... it will ask for all the details and set everything up for you then it will bring up a browser window to login. Doesn't matter if you just copy and paste a force.json, you will always need to login with the browser first.

**Comment by julesx** [comment]
i was hoping i could copy force.json over so i wouldn't have to re-setup forcecode in every workspace, but that's not a big deal.

it's more that it just doesn't work at all on first launch :)

**Comment by daishi4u** [comment]
OK, I think it looks fine. As far as startup goes, ForceCode relies on 3 salesforce extensions for code completion. The apex one is what creates that folder and file and is what causes the start lag because ForceCode has to wait for those extensions to start before it ends. There's nothing I can do about that part.

**Comment by julesx** [comment]
just a note: i realized when creating my vs code workspace i wasn't copying force.json to the root folder.  i fixed this, and now when i first start vs, i guess it thinks it has previously logged in, so stuff fails.  i have to open force code menu, explicitly log out, then restart vs code and then i am prompted to login properly.

**Comment by daishi4u** [comment]
So, I just did a test. I copied my force.json from one folder to another, and ONLY this file. Then i opened the folder in vscode and waited. ForceCode started up after a little bit and I was already logged in. When you first opened these fresh projects, did you wait for it to start or just try the open menu command right away? Because if you did then I'm thinking that may be the issue. Patience, it must wait for the other salesforce extensions to load! :)
